### PR TITLE
fix: incorrect lint error position

### LINKED ISSUE #16524: [Bug]: Invalid lint position when same string segment is present in and outside the curly braces
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Screenshot 2022-09-05 at 07 59 45

### Steps To Reproduce

1. Go to any editor field
2. type` has {{hss}}` (basically, same value outside the binding bracket and inside it )
3. Observe lint error position

### Public Sample App

_No response_

### Version

Cloud

### PR DESCRIPTION
**Pull request by ohansFavour**
## Description

Fixes #16524 

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
Jest

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/client/src/components/editorComponents/CodeEditor/lintHelpers.ts [hunk 1]
```diff
@@ -168,7 +174,12 @@ export const getLintAnnotations = (
         const currentLine = bindingLocation.line + line;
         const lineContent = lines[currentLine] || "";
         const currentCh =
-          bindingLocation.line !== currentLine ? ch : bindingLocation.ch + ch;
+          bindingLocation.line !== currentLine
+            ? ch
+            : bindingLocation.ch +
+              ch +
+              // Add 2 to account for "{{", if binding is a dynamicValue
+              (isDynamicValue(originalBinding) ? 2 : 0);
```
Comment (Rishabh-Rathod) [inline app/client/src/components/editorComponents/CodeEditor/lintHelpers.ts:182]: nit: can we convert it to a conditional if statement to make it more readable?

**Review by Rishabh-Rathod — APPROVED** [review]
LGTM.

**Comment by ramsaptami** [comment]
Verified fix on DP and it's gtg to release 
CC: @ohansFavour

**Review by Rishabh-Rathod — APPROVED** [review]

**Pull request merged**